### PR TITLE
split out live call negotiation

### PR DESCRIPTION
**Pull request by koush**

### CODE REVIEW

File: api/peer-connection.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export class PeerConnection {
           ],
         },
       })),
-      audioTransceiver = pc.addTransceiver(this.returnAudioTrack, {
+      audioTransceiver = pc.addTransceiver('audio', {
```
Comment (dgreif) [inline api/peer-connection.ts:49]: hmm was this change needed?

Comment (koush) [reply api/peer-connection.ts:49]: hm no I had this for testing my weird connection drops and forgot to take it out

**Comment by koush** [comment]
alright the latest is merged and changes minimized. didn't realize you had already removed sip from the library.

**Comment by koush** [comment]
was testing this endpoint and found that it is returning a sdp that seems to be specifically for clients that can handle H264 high. Presumably this is because they know/think the client is android. in any casem it fails on safari and chromecast, etc. I'll readd the original endpoint I had before as a fallback option for browsers.

**Comment by koush** [comment]
Confirmed that reintroducing the browser endpoints resolved my Safari/Chromecast issues. https://github.com/koush/scrypted/commit/9850eebdda6a9b909acb5f55916b92e56b551806

**Comment by dgreif** [comment]
I've added the ability to pass in a custom `createPeerConnection` option in the `beta` branch, so this should no longer be needed. Thanks for continuing to work with me on this api!
